Sources/Starknet/Accounts/StarknetAccountProtocol.swift

```diff
@@ -80,98 +150,258 @@ public protocol StarknetAccountProtocol {
     }
 
-    /// Estimate fee for a call
+    /// Estimate fee for a call as invoke transaction v1
     ///
     /// - Parameters:
     ///  - call: a call for which the fee should be estimated.
-    ///  - nonce: a nonce to be used in a transaction
+    ///  - nonce: a nonce to be used in a transaction.
+    ///  - skipValidate: flag indicating whether validation of the transaction should be skipped.
+    ///
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV1(call: StarknetCall, nonce: Felt, skipValidate: Bool = true) async throws -> StarknetFeeEstimate {
+        try await estimateFeeV1(calls: [call], nonce: nonce, skipValidate: skipValidate)
+    }
+
+    /// Estimate fee for a call as invoke transaction v3
+    ///
+    /// - Parameters:
+    ///  - call: a call for which the fee should be estimated.
+    ///  - nonce: a nonce to be used in a transaction.
+    ///  - skipValidate: flag indicating whether validation of the transaction should be skipped.
+    ///
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV3(call: StarknetCall, nonce: Felt, skipValidate: Bool = true) async throws -> StarknetFeeEstimate {
+        try await estimateFeeV3(calls: [call], nonce: nonce, skipValidate: skipValidate)
+    }
+
+    /// Estimate fee for a call as invoke transaction v1
+    ///
+    /// - Parameters:
+    ///  - call: a call for which the fee should be estimated.
+    ///  - skipValidate: flag indicating whether validation of the transaction should be skipped.
+    ///
     /// - Returns: struct containing fee estimate
-    func estimateFee(call: StarknetCall, nonce: Felt) async throws -> StarknetFeeEstimate {
-        try await estimateFee(calls: [call], nonce: nonce)
+    func estimateFeeV1(call: StarknetCall, skipValidate: Bool = true) async throws -> StarknetFeeEstimate {
+        try await estimateFeeV1(calls: [call], skipValidate: skipValidate)
     }
 
-    /// Estimate fee for a call
+    /// Estimate fee for a call as invoke transaction v3
     ///
     /// - Parameters:
     ///  - call: a call for which the fee should be estimated.
+    ///  - skipValidate: flag indicating whether validation of the transaction should be skipped.
+    ///
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV3(call: StarknetCall, skipValidate: Bool = true) async throws -> StarknetFeeEstimate {
+        try await estimateFeeV3(calls: [call], skipValidate: skipValidate)
+    }
+
+    /// Estimate fee for a deploy account transaction v1
+    ///
+    /// - Parameters:
+    ///  - classHash: class hash of account to be deployed
+    ///  - calldata: constructor calldata
+    ///  - salt: contract salt
+    ///  - nonce: nonce of the account to be deployed
+    ///
+    /// - Returns: struct containing fee estimate
+    func estimateDeployAccountFeeV1(classHash: Felt, calldata: StarknetCalldata, salt: Felt, nonce: Felt) async throws -> StarknetFeeEstimate {
```

Comment:
We have an overload in jvm that sets nonce to zero by default. Should we also add it here, both for v1 and v3, with nonce = 0 or obtained via `getNonce`?

Reply:
I don't have a problem with leaving it as default value, but leave the final decision up to you